R/chat.R

```diff
@@ -412,22 +448,24 @@ ChatOpenAI <- R6::R6Class("ChatOpenAI",
 #' @export
 print.ChatOpenAI <- function(x, ...) {
   cat("<ChatOpenAI>\n")
-  messages <- x$.__enclos_env__$private$messages
-  for (message in messages) {
+  for (message in x$messages()) {
     color <- switch(message$role,
-      user = "blue",
-      system = ,
-      assistant = "green"
+      user = cli::col_blue,
+      system = identity,
+      assistant = cli::col_green
     )
-    cat(cli::rule(message$role, col = color), "\n", sep = "")
-    cat(message$content, "\n", sep = "")
+    cli::cli_rule("{color(message$role)}")
+    # Using cli_text for word wrapping. Passing `"{message$content}"` instead of
+    # `message$content` to avoid evaluation of the (potentially malicious)
+    # content.
+    cli::cli_text("{message$content}")
```

Comment:
This change emits the chat history as messages instead of catting to stdout. Is that wrong for a `print` method? I wanted to use `cli_text` for the word wrapping, but can do that via strwrap if we want the history to go to stdout.

Reply:
It's a litte wrong, but I don't think it's worth trying to do it 100% correctly.